via/static/scripts/video_player/components/VideoPlayerApp.tsx

```diff
@@ -158,46 +169,40 @@ export default function VideoPlayerApp({
   };
 
   return (
-    <div className="w-full flex">
-      <div
-        className={classnames(
-          // This column will grow in width per the parent flex container and
-          // allow the contained media (video) to scale with available space.
-          // The column flex layout established here ensures this container fills
-          // the full height of the parent flex container
-          'grow flex flex-col p-3'
-        )}
-      >
-        <YouTubeVideoPlayer
-          videoId={videoId}
-          play={playing}
-          time={timestamp}
-          onPlayingChanged={setPlaying}
-          onTimeChanged={setTimestamp}
-        />
-      </div>
+    <div
+      data-testid="app-container"
+      className={classnames(
+        // Constrain height of interface to viewport height
+        'flex flex-col h-[100vh] min-h-0'
+      )}
+    >
       <div
+        data-testid="top-bar"
         className={classnames(
-          // Full-height column with a width allowing comfortable line lengths
-          'h-[100vh] w-[450px] flex flex-col',
-          'bg-grey-0 border-x',
-          // TODO: This is a stopgap measure to prevent controls from being
-          // interfered with (overlaid) by sidebar controls and toolbar
-          'mr-[30px]'
+          // Match height of sidebar's top-bar
+          'h-[40px] min-h-[40px]',
+          // Background is one shade darker than sidebar top-bar
+          'w-full flex items-center gap-x-3 px-2 bg-grey-0 border-b'
         )}
       >
-        <div
-          className={classnames(
-            'p-1 bg-grey-1',
-            // TODO: This is a stopgap measure to prevent the right side of the
-            // search input from being inpinged on by sidebar controls
-            'pr-2'
-          )}
-          data-testid="search-bar"
-        >
+        <LogoIcon />
+        <div data-testid="filter-container" className="grow text-right">
           <Input
-            aria-label="Transcript filter"
             data-testid="filter-input"
+            aria-label="Transcript filter"
+            classes={classnames(
+              // Match height of search input in sidebar
+              'h-[32px]',
+              // TODO: Temporary prevention of sidebar controls overlapping
+              'mr-[22px]',
+              {
+                // Adapt width to match width of transcript
```

Comment:
If I can find a straightforward way to abstract some of this away in a future iteration, I will. I tried a couple of abortive things today but felt in danger of imposing premature abstraction. So for now please have patience with verbose styling until we get into more of a "clean-up" mode.